### PR TITLE
Remove Web3.js From 0x-monorepo 🍾 

### PR DESCRIPTION
**Pull request by fabioberger**
## Description

This PR converts the remaining calls to `Web3.js` to use our `_sendRawPayloadAsync` method. Additionally, it also removes our reliance on `Web3`'s formatter for marshalling and unmarshalling JSON RPC payloads. We now do this ourselves.

This PR should reduce 0x.js bundle size by at least 8MB, and also reduce the installation time of the package drastically.

## Testing instructions

Run all the tests (includes additional test coverage for the web-wrapper package).

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [x] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/assert/src/index.ts [hunk 1]
```diff
@@ -41,8 +41,8 @@ export const assert = {
         value: string,
         stringEnum: any /* There is no base type for every string enum */,
     ): void {
-        const doesBelongToStringEnum = !_.isUndefined(stringEnum[value]);
-        const enumValues = _.keys(stringEnum);
+        const enumValues = _.values(stringEnum);
+        const doesBelongToStringEnum = _.includes(enumValues, value);
```
Comment (fabioberger) [inline packages/assert/src/index.ts]: Our assertion was previously testing the enum keys, not values (which is fine when they are the same... but not when they are not).

File: packages/base-contract/src/index.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export class BaseContract {
             // Awaiting https://github.com/Microsoft/TypeScript/pull/13288 to be merged
         } as any;
         if (_.isUndefined(txDataWithDefaults.gas) && !_.isUndefined(estimateGasAsync)) {
-            txDataWithDefaults.gas = await estimateGasAsync(txData);
+            txDataWithDefaults.gas = await estimateGasAsync(txDataWithDefaults as any);
```
Comment (fabioberger) [inline packages/base-contract/src/index.ts]: We need to be passing in `from` to `estimateGasAsync`. @LogvinovLeon, do you see any issues with this change?

**Comment by LogvinovLeon** [comment]
8 MB? What's the current bundle size? Can you attach webpack snapshots? I have hard time believing it.

File: packages/base-contract/src/index.ts (hunk 2 above)
Comment (fabioberger) [reply packages/base-contract/src/index.ts]: This is actually not true, we only need `from` when sending transactions, not when estimating gas. Reverted this change.

**Comment by LogvinovLeon** [comment]
LGTM after the web3-wrapper CHANGELOG is updated

**Pull request merged**